drivers/dome/ddw_dome.cpp

```diff
@@ -0,0 +1,560 @@
+
+void ISPoll(void *p);
+
+void ISGetProperties(const char *dev)
+{
+    ddw->ISGetProperties(dev);
+}
+
+void ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    ddw->ISNewSwitch(dev, name, states, names, n);
+}
+
+void ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n)
+{
+    ddw->ISNewText(dev, name, texts, names, n);
+}
+
+void ISNewNumber(const char *dev, const char *name, double values[], char *names[], int n)
+{
+    ddw->ISNewNumber(dev, name, values, names, n);
+}
+
+void ISNewBLOB(const char *dev, const char *name, int sizes[], int blobsizes[], char *blobs[], char *formats[],
+               char *names[], int n)
+{
+    INDI_UNUSED(dev);
+    INDI_UNUSED(name);
+    INDI_UNUSED(sizes);
+    INDI_UNUSED(blobsizes);
+    INDI_UNUSED(blobs);
+    INDI_UNUSED(formats);
+    INDI_UNUSED(names);
+    INDI_UNUSED(n);
+}
+
+void ISSnoopDevice(XMLEle *root)
+{
+    ddw->ISSnoopDevice(root);
+}
+
+DDW::DDW()
+{
+    setVersion(1, 0);
+    SetDomeCapability(DOME_CAN_ABORT | DOME_CAN_ABS_MOVE | DOME_CAN_PARK | DOME_HAS_SHUTTER);
+}
+
+bool DDW::initProperties()
+{
+    INDI::Dome::initProperties();
+
+    IUFillNumber(&FirmwareVersionN[0], "VERSION", "Version", "%2.0f", 0.0, 99.0, 1.0, 0.0);
+    IUFillNumberVector(&FirmwareVersionNP, FirmwareVersionN, 2, getDeviceName(), "FIRMWARE", "Firmware", INFO_TAB,
+                       IP_RO, 60, IPS_IDLE);
+
+    SetParkDataType(PARK_AZ);
+
+    addAuxControls();
+
+    // Set serial parameters
+    serialConnection->setDefaultBaudRate(Connection::Serial::B_9600);
+
+    setPollingPeriodRange(1000, 3000); // Device doesn't like too long interval
+    setDefaultPollingPeriod(1000);
+    return true;
+}
+
+/************************************************************************************
+ *
+* ***********************************************************************************/
+bool DDW::SetupParms()
+{
+    if (InitPark())
+    {
+        // If loading parking data is successful, we just set the default parking
+        // values.
+        SetAxis1ParkDefault(0);
+    }
+    else
+    {
+        // Otherwise, we set all parking data to default in case no parking data is
+        // found.
+        SetAxis1Park(0);
+        SetAxis1ParkDefault(0);
+    }
+
+    FirmwareVersionN[0].value = fwVersion;
+    FirmwareVersionNP.s       = IPS_OK;
+    IDSetNumber(&FirmwareVersionNP, nullptr);
+    return true;
+}
+
+/************************************************************************************
+ *
+* ***********************************************************************************/
+bool DDW::Handshake()
+{
+    // Send GINF command and check if the controller responds
+    int rc = writeCmd("GINF");
+    if (rc != TTY_OK)
+        return false;
+
+    std::string response;
+    rc = readStatus(response);
+    if (rc != TTY_OK)
+        return false;
+
+    LOGF_DEBUG("Initial response: %s", response.c_str());
```

Comment:
Have you considered using [sendCommand](https://github.com/indilib/indi/blob/fc26c0129693a0c9390f55d7fe71ffbb0ad3176f/drivers/dome/domepro2.h#L126
) function? It is generic enough to cover for most use cases.

Reply:
It might work for this handshake case, normally the device doesn't immediately respond at all, but might also send random responses to hand controller events so they must be polled separately instead of simple command/response kind of protocol. Again this was partially leftover from ScopeDome driver, which uses a binary protocol with checksums etc.

Reply:
Ok I see, alright let's stick with what works now.